### PR TITLE
Add ENTRYPOINT override to Dockerfile

### PR DESCRIPTION
**Pull request by tverbeiren**
See #282

### CODE REVIEW

**Comment by tverbeiren** [comment]
_Caveat_: does not yet do what it's supposed to do in my tests locally!

File: src/main/scala/io/viash/platforms/DockerPlatform.scala [hunk 1]
```diff
@@ -156,7 +156,16 @@ case class DockerPlatform(
 
   @description("Additional requirements specific for running unit tests.")
   @since("Viash 0.5.13")
-  test_setup: List[Requirements] = Nil
+  test_setup: List[Requirements] = Nil,
+
+  @description("Override the entrypoint of the base container. Default set `ENTRYPOINT []`.")
+  @since("Viash 0.7.4")
+  entrypoint: Option[String] = Some("[]"),
```
Comment (rcannood) [inline src/main/scala/io/viash/platforms/DockerPlatform.scala:163]: Would this work?

```suggestion
  entrypoint: Option[List[String]] = Some(Nil),
```

Does this allow to differentiate between `None` (no entrypoint should be added) and `Some(Nil)` (entrypoint should be `[]`)

Comment (Grifs) [reply src/main/scala/io/viash/platforms/DockerPlatform.scala]: I changed it to `entrypoint: Option[Either[String, List[String]]] = Some(Right(Nil)),` as it allows [the full described functionality](https://docs.docker.com/engine/reference/builder/#entrypoint) of either single string or an array of strings.
The single string is a non-preferred option but the actual additional code to support this is quite minimal.
The biggest issue was the distinction between a null value and an empty string.

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,8 @@
   - `{output}`: path to the destination directory when building the component
   - `{abs-output}`: absolute path to the destination directory when building the component
 
+* `docker platform`: Add `entrypoint` and `cmd` parameters to the docker platform config that allows overriding the default docker container settings.
```
Comment (rcannood) [inline CHANGELOG.md:19]:
```suggestion
* `DockerPlatform`: Add `entrypoint` and `cmd` parameters to the docker platform config that allows overriding the default docker container settings (PR #432).

**Review by Grifs — APPROVED** [review]

**Pull request merged**